packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -476,11 +494,20 @@ export class VpcSubnet extends VpcSubnetRef {
     constructor(parent: cdk.Construct, name: string, props: VpcSubnetProps) {
         super(parent, name);
         this.availabilityZone = props.availabilityZone;
+        if (props.tags !== undefined) {
+            this.tags = props.tags.map( tag => new cdk.Token(tag) );
+            if (props.tags.filter( tag => tag.key === 'Name' ).length !== 1) {
+                this.tags.push(new cdk.Token({key: 'Name', value: name}));
```

Comment:
Use `this.path` instead of `name` as the default. `name` is a local logical identity and path is globally unique within the stack, and provides much more context.